### PR TITLE
Topic-progressDialog

### PR DESCRIPTION
**Pull request by tlwalla1**
@jbest84 @CHenrie @jeremyphillips9998 Please Review. Briefcasing now shows a busy indicator and a subsequent dialog to prompt the user to navigate to the briefcase view.

### CODE REVIEW

File: src/BusyIndicator.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  start: function start() {
+    this._busyDeferred = new Deferred();
+    this._busyDeferred.then(this.complete.bind(this));
+
+    const indicator = domConstruct.toDom(this.busyIndicatorTemplate.apply(this));
+    domConstruct.place(indicator, this.busyIndicatorNode);
+    this._busyIndicator = this.busyIndicatorNode.children[0];
+    domClass.add(this._busyIndicator, 'busyIndicator--active');
+
+    return this._busyDeferred;
```
Comment (jbest84) [inline src/BusyIndicator.js]: The return should be `this._busyDeferred.promise;`

_EDIT_ - It appears outside code is resolving this promise. I would define a finish method on this to resolve this._busyDeferred and not rely on the calling code to do so. The underlying deferred should not be exposed to calling code, just the promise API which does not include a resolve.

File: src/Offline/_DetailOfflineMixin.js [hunk 2]
```diff
@@ -46,7 +57,25 @@ export default declare('argos.Offline._DetailOfflineMixin', null, {
       viewId: this.id,
     };
     OfflineManager.briefCaseEntity(entityName, entityId, options).then((result) => {
-      this.onEntityBriefcased(result);
+      // Show complete modal dialog
+      App.modal.disableClose = false;
+      App.modal.showToolbar = true;
+      loadingModal.resolve(true);
+      busyDeferred.resolve(true);
```
Comment (jbest84) [inline src/Offline/_DetailOfflineMixin.js]: `busyIndicator.finish()`?

File: localization/locales/argos/en/strings.l20n [hunk 3]
```diff
@@ -244,3 +244,21 @@
...
 >
+
+<busyIndicator ""
+  loadingText: "Loading"
+>
+
+<_listOfflineMixin
+  cancelText: "Cancel"
+  okayText: "Okay"
+  completeText: "Complete"
```
Comment (CHenrie) [inline localization/locales/argos/en/strings.l20n]: Why are we not creating a dialog that has the localization strings

**Comment by tlwalla1** [comment]
@jbest84 @CHenrie I added in a progress bar that uses deferred.progress() to update the width. This is used in the list briefcasing to better show the user the progress (since it is fairly long on large lists).

**Pull request merged**